src/software/new_geom/util/intersection.cpp

```diff
@@ -29,7 +15,7 @@ std::optional<Point> intersection(const Point &a, const Point &b, const Point &c
     double y4 = d.y();
 
     double denominator = (x1 - x2) * (y3 - y4) - (y1 - y2) * (x3 - x4);
-    if (denominator == 0)
+    if (std::fabs(denominator) < GeomConstants::EPSILON)
```

Comment:
Should we use the new `almostEqual(...)` function here?

Reply:
Fixed, I'll probably tackle the rest of `double` comparisons in geom in another pr